### PR TITLE
Separate loadouts from settings

### PR DESCRIPTION
**Pull request by stixes**
Updated a bunch of stuff, and some cleanup and refactoring in the mix aswell.. Sorry about the mess :D 

Major points:
- Settings now in 2 files with different lifecycle:
-- settings.json, with app settings and automatic save on exit (hotkeys etc.)
-- loadouts.json , with the loadout setup. No automatic save, but a warning on exit if anything was updated.
-- The code for this was unified to non-singleton, and are now available under the model, or via a helper function in the controller.

### CODE REVIEW

File: src/loadouts.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    This class is used primarily in the settings.
+    """
+    def __init__(self, name, macroKeys):
+        self.name = name
+        self.macroKeys = macroKeys
+
+class LoadoutManager:
+    def __init__(self):
+        self._observers = []
+        self.loadouts = {utilities.generateUuid(): Loadout("Loadout 1", {"1": "1"})}
```
Comment (Noloxs) [inline src/loadouts.py:22]: We should take this opportunity to design some more meaningful default loadouts
Maybe just 2 loadouts: Bugs, Automatons with ~5 stratagems each

File: src/settings.py [hunk 2]
```diff
@@ -155,12 +131,10 @@ def migrate_2_to_3(self):
...
-        self.name = name
-        self.macroKeys = macroKeys
\ No newline at end of file
+    def migrate_3_to_4(self):
+        self.version = 4
+    
+        if hasattr(self, 'loadouts'):
+            delattr(self, 'loadouts')
+
+        print("Settings migrated to version 4. Remember to save.")
```
Comment (Noloxs) [inline src/settings.py:140]: Should we add migration to the loadouts file?

**Review by Noloxs — COMMENTED** [review]
Looks good so far.
Looking forward to testing it out

File: src/loadouts.py (hunk 1 above)
Comment (stixes) [reply src/loadouts.py]: This has been addressed

File: src/settings.py (hunk 2 above)
Comment (stixes) [reply src/settings.py:143]: Can use same as settings, however, until it is needed, it would just be messy.. so omitted.

**Comment by stixes** [comment]
Actually ready for testing now :D

**Review by Noloxs — CHANGES_REQUESTED** [review]
**Issues:**
- Unable to change "Open stratagem list" binding
- Unable to change: Trigger delay, Jitter etc.
- Changing always on Top doesn't activate before restart of hell snake
- Line 85:  AttributeError: 'PynputKeyListener' object has no attribute 'key_release_handlers' (not initialized, happens when currentLoadoutId doesn't match)
- Unable to open edit loadouts:
  - edit_loadout_dialog.py", line 178, in update_macros
    self.editMacros.update({key:self.controller.model.stratagems[stratagemId]})

---    

**Issues unrelated to PR:**
- Description is not reset when changing executer (Connect to arduino and set executor to pynput, then it will still say "Connected to: ...")
- In pico passthrough boot.py file. usb_cdc.enable(console=True, data=True) should be usb_cdc.enable(console=False, data=True) to fix issue with double entries from same device

**Comment by Noloxs** [comment]
I found a way to replicate the PynputListerner issue.

NB: I was also able to confirm that this issue breaks the binding mechanism making it impossible to rebind keys.

To replicate:

1. Start hell snake
2. Spam keys before it hell snake initializes

**Comment by Noloxs** [comment]
I have created issues for the remaining issues unrelated to the PR

**Review by Noloxs — APPROVED** [review]
Looks good. Merge when happy 👍

**Pull request merged**